timely_beliefs/beliefs/classes.py

```diff
@@ -419,6 +430,28 @@ def search_session(  # noqa: C901  # todo: remove after removing deprecated argu
             source_cls = sources[0].__class__
             q = q.join(source_cls).filter(cls.source_id.in_([s.id for s in sources]))
 
+        # Apply most recent beliefs filter
+        if most_recent_only:
+            subq = (
+                session.query(
+                    cls.event_start,
+                    cls.source_id,
+                    func.min(cls.belief_horizon).label("most_recent_belief_horizon"),
+                )
+                .group_by(cls.event_start, cls.source_id)
+                .order_by(func.min(cls.belief_horizon).desc())
```

Comment:
I'm not sure why this `order_by` clause is needed. We should have one belief horizon per event&source combination by now.